### PR TITLE
Add logout logic

### LINKED ISSUE #436: Add SignOutTapped action
## Description
Add SignOutTapped action, `SignOut` effect and `SignOutCompleted` action. 
SignOut effect and the SignOutCompleted action are the most interesting part here since it should change the state in a way that it takes the user to the login view and triggers a clean up of the data like TEs etc. and api tokes and what not. Whatever is not possible to implement at this point, leave a comment and make an issue so that it can be handled later on.

As for the correct name `SignOut` vs `Log Out` I leave it to the dev working on this to talk with product to find out what is best.

Check [notion](https://www.notion.so/toggl/Settings) for details regarding parameters and effects

## Dependencies
#417
## Definition of done
- [ ] Add SignOutTapped action
- [ ] Add SignOut effect
- [ ] Add SignOutCompleted action
- [ ] Add tests

### PR DESCRIPTION
**Pull request by Odrakir**
## Info
Adds actions, reducer logic and service methods to successfully logout user.
### Technical considerations
I decided to move this to the `AppReducer`, as it makes sense to take it as a global level action even if it's started in Settings (although it could also be started as a response to an API error, so that's more reason to move it there). It might need access to a lot of services and it needs to reset most (if not all) the state

I'm willing to hear other thoughts though.

Also, this is not clearing everything we'll need to clear in the end, we'll need to add more stuff to the reducer as needed.

### Relationships
Closes #436 

### Pull Request checklist
- [ ] The code does not hinder the app's performance
- [ ] The code does not introduce regressions
- [ ] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests

### Pointers
Tiny commits are easy to review commits

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Do it

### CODE REVIEW

File: Modules/OtherServices/OtherServices/UserDefaultService.swift [hunk 1]
```diff
@@ -60,4 +64,9 @@ public class UserDefaultService {
 
     @UserDefault(UserDefaultKey.calendarPermissionWasGranted, defaultValue: false)
     public var calendarPermissionWasGranted: Bool
+
+    public func clear() {
+        _useCalendarIntegration.clear()
+        _selectedUserCalendarIds.clear()
```
Comment (sofokli1) [inline Modules/OtherServices/OtherServices/UserDefaultService.swift:70]: why not the `calendarPermissionWasGranted`?

Comment (Odrakir) [reply Modules/OtherServices/OtherServices/UserDefaultService.swift:70]: I'm not sure about that one, to be honest. We can't clear the system status on that permission, so I guess we have to still keep track of wether the user gave permission already or not.

Comment (sofokli1) [reply Modules/OtherServices/OtherServices/UserDefaultService.swift:70]: I guess we can change it later if needed

File: Modules/Database/Database/CoreData/CoreDataModel.swift [hunk 2]
```diff
@@ -62,8 +62,16 @@ extension CoreDataModel {
     static func deleteAll(
         in context: NSManagedObjectContext
     ) throws {
-        let fetchRequest: NSFetchRequest<NSFetchRequestResult> = NSFetchRequest(entityName: ManagedObject.entityName)
-        let deleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
-        try context.persistentStoreCoordinator?.execute(deleteRequest, with: context)
+        #if DEBUG //Batch delete doesn't work with the in-memory store used in tests
```
Comment (sofokli1) [inline Modules/Database/Database/CoreData/CoreDataModel.swift:65]: isn't this going to be slow when testing logout in simulator/device during development?

Comment (Odrakir) [reply Modules/Database/Database/CoreData/CoreDataModel.swift:65]: I don't think it will, because it will happen in memory, but we'll keep an eye on that.

**Pull request merged**